### PR TITLE
Possums and Raccoons and Ferrets, Oh My!

### PR DESCRIPTION
**Pull request by Willhelm53**
## About the PR 

Adds Possums, Raccoons, and Ferrets using sprites from Goonstation. They are all pettable (with the animals' consent of course!) Possums will hiss at you when they don't want to be pet (just like in real life 😭), but at least they won't give you rabies until someone codes it. Ferrets will make happy ferret noises when you pet them. All sound effects used are creative-commons licensed as specified in the license.txt.

In a nod to Goonstation, a named possum called Morty has also been added, however, he won't grace the Morgue's presence without a little help from our mappers. Coming soon, we can hope!

There is now an SS14-exclusive named raccoon called Morticia. Just like her Addams Family namesake, she is a powerful creature of the night whose eyeshadow is always on point.

(The prophecy from #6530 has been fulfilled!)

**Screenshots**

image

**Changelog**

:cl:
- add: Added Possums, Raccoons and Ferrets, with a couple of new sound effects.
- add: Added a named Possum (Morty) and Raccoon (Morticia)

### CODE REVIEW

**Comment by Peptide90** [comment]
Another 10/10 Willhelm PR coming at ya (But oh my god my eyes that branch name man. Is that payback for the meta.json I sent in discord?)

**Comment by Willhelm53** [comment]
> Another 10/10 Willhelm PR coming at ya (But oh my god my eyes that branch name man. Is that payback for the meta.json I sent in discord?)

~~Yes.~~ Every time you complain, I add an additional animal to the game and thus make the branch name longer 😁

**Comment by Willhelm53** [comment]
>tfw I'm only one line plus some formatting away from having that sweet, sweet "No C#" tag 😭

**Comment by Matz05** [comment]
Interesting. I assume these possums are Virginia Opossums (the 'possums' found in the USA)? Apparently, they're actually strongly resistant to rabies (something about a very low body temperature, or possibly being a marsupial?), and are probably the least likely of wild warm-blooded animals to spread it to you -- though open-mouthed hissing and drooling in their intimidation display makes them often LOOK rabid even when they're not.

**Comment by Willhelm53** [comment]
> Interesting. I assume these possums are Virginia Opossums (the 'possums' found in the USA)? 

Yeppers! Way ahead of ya 😄 

image

> Apparently, they're actually strongly resistant to rabies

Interesting! I hadn't known this - and there I go perpetuating a harmful stereotype about our friend-shaped little fellas. Thank you for saying! I retract my bit about the rabies, although it'd be interesting to see what other animals on the station could be made rabid in the future 👀

**Comment by Emisse** [comment]
can you add spawners for them so we can add them into maps at round start. I feel like giving the quartermaster a pet raccoon might be fitting for them lol

**Comment by Willhelm53** [comment]
>can you add spawners for them so we can add them into maps at round start

I will for sure be looking into how to do that this weekend! Great idea on the QM raccoon. Maybe that's where Morticia can go!

File: Content.Server/Interaction/InteractionPopupSystem.cs [hunk 1]
```diff
@@ -61,10 +61,13 @@ private void OnInteractHand(EntityUid uid, InteractionPopupComponent component,
         else
             _popupSystem.PopupEntity(msg, uid, Filter.Entities(args.User)); //play only for the initiating entity.
 
-        if (component.SoundPerceivedByOthers)
-            SoundSystem.Play(Filter.Pvs(args.Target), sfx, args.Target); //play for everyone in range
-        else
-            SoundSystem.Play(Filter.Entities(args.User, args.Target), sfx, args.Target); //play only for the initiating entity and its target.
+        if (sfx != "") //not all cases will have sound.
```
Comment (mirrorcult) [inline Content.Server/Interaction/InteractionPopupSystem.cs:64]:
```suggestion
        if (sfx != string.Empty) //not all cases will have sound.
```

or make it nullable, that would probably be preferable

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 2]
```diff
@@ -949,3 +949,119 @@
...
+- type: entity
+  name: raccoon
+  parent: SimpleMobBase
+  id: MobRaccoon
+  description: Trash panda!
+  components:
+  - type: Sprite
+    drawdepth: Mobs
+    sprite: Mobs/Animals/raccoon.rsi
+    scale: 0.7, 0.7
```
Comment (mirrorcult) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:1021]: not sure how i feel about them being directly scaled down, but  i guess it looks alright?

**Review by mirrorcult — APPROVED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 2 above)
Comment (Willhelm53) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:1021]: As far as I can see, we don't have an alternative apart from respriting the dang things from scratch.

I believe scaling down the height is all we do for our `Dwarf` playable race currently.

File: Content.Server/Interaction/InteractionPopupSystem.cs (hunk 1 above)
Comment (Willhelm53) [reply Content.Server/Interaction/InteractionPopupSystem.cs]: Made it nullable on latest commit. Would you please make sure I've done so correctly? I've tested on local and it still runs fine.

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 2 above)
Comment (mirrorcult) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:1021]: aight

**Pull request merged**